Lib/idlelib/pyparse.py

```diff
@@ -101,46 +100,19 @@
 """, re.VERBOSE).match
 
 
-class StringTranslatePseudoMapping(Mapping):
-    r"""Utility class to be used with str.translate()
+class ParseMap(dict):
```

Comment:
Isn't this class exactly equivalent to `defaultdict(lambda:"x")`

Reply:
This class doesn't add the keys to the dictionary when they haven't been found.  It will return the default value without growing the dictionary.

Reply:
Also, defaultdict calls the function to get 'x', instead of just returning it.